department-of-veterans-affairs/va.gov-cms
# [WIP] VACMS 8299 + 8300 site alert improvements

## Description
Closes #8299 and #8300. Contributes & uses a patch for adding dismissible behaviors to site alert. Creates site alert component in design system, used in Drupal.

All site alert forms now have a "dismissible" checkbox that allows them to be dismissed. This is set to false by default.
Dismissed alerts are remembered in sessionstorage, so it will not re-appear for a user on any page once dismissed. If the dismissed site alert is saved again as no longer dismissible, it will re-appear as expected.

Some minor styling work needed still, dependent on #8994 to clean up some SB structure so I will revisit this later this sprint. 

## Testing done
added multiple alerts,  some dismissible and not, dismissed alerts, checked different pages and verified alerts still disappeared (and did not re-appear on the ajax refreshes)

## Screenshots


## QA steps
- as admin, log in and navigate to `/admin/config/system/site-alerts`
- enable any alert
  - check the schedule to make sure it will actually appear while you're testing
  - warning that any "High" status alerts will block the full screen unless they are marked dismissible! Need to delete the html element if you accidentally lock yourself out
  - save the alert
- create another alert that is dismissible, save it
- see alerts show on next page, see that one can be dismissed and one cannot.
- dismiss the dismissible alert
- navigate to a new page (or refresh), see that the dismissed alert does not reappear

- log out, log in as any editor (kitty.walsh@va.gov)
- see the alerts behave as expected.... you may need to delete the cookie in session storage labeled "siteAlertsVisibility" to reset all alert status for your browser, or use an incognito window

### Definition of Done


### Select Team for PR review



### Is this PR blocked by another PR?
